src/main/java/com/google/api/generator/engine/format/JavaDocComment.java

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.api.generator.engine.format;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.googlejavaformat.java.Formatter;
+import java.util.Optional;
+
+@AutoValue
+public abstract class JavaDocComment implements Comment {
+  public abstract ImmutableList<String> comments();
+
+  public abstract Optional<String> deprecatedText();
+
+  public abstract Optional<String> sampleCode();
+
+  public abstract Optional<String> throwsText();
+
+  public abstract ImmutableList<ParamPair> params();
```

Comment:
What is the motivation for using a list of custom pair structs over a Map?

Reply:
addParams() only accepts 0/1 argument, so I made a class for the param pair, but correct me if I'm wrong, this bit me during the implementation. thx!

Reply:
ImmutableMap is used over a custom pair.